src/Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs

```diff
@@ -116,6 +117,12 @@ private PowerShell CreatePowerShell()
             return pwsh;
         }
 
+        public override IReadOnlyCollection<string> GetVariableNames()
+        {
+            // FIX: (GetVariableNames) 
+            return Array.Empty<string>();
```

Comment:
@TylerLeonhardt Do you know a way to do get the available variable names?

Reply:
```csharp
var ec = pwsh.Runspace.SessionStateProxy.PSVariable.Get("ExecutionContext");
ec.InvokeProvider.Item.Get("variable:\*").Select(v => v.Name);
```

Something like this should work. I feel like there's a one liner... but that's the best I can think of.

Keep in mind, this API is _not_ thread safe. It requires the `pwsh.Runspace` object to not have anything running in it.

We might be getting _really_ close to needing a Semaphore around `pwsh`.

Reply:
Got the one liner:

```csharp
pwsh.Runspace.SessionStateProxy.InvokeProvider​.​Item​.​Get​(​"​variable:​\*​"​).Select(v => v.Name)
```